grudge/array_context.py

```diff
@@ -509,11 +510,34 @@ class PytestPyOpenCLArrayContextFactory(
         _PytestPyOpenCLArrayContextFactoryWithClass):
     actx_class = PyOpenCLArrayContext
 
+    def __call__(self):
+        actx = super().__call__()
+        if actx.allocator is not None:
+            return actx
+
+        from pyopencl.tools import ImmediateAllocator, MemoryPool
+        alloc = MemoryPool(ImmediateAllocator(actx.queue))
+
+        return self.actx_class(
+                actx.queue,
+                allocator=alloc,
+                force_device_scalars=self.force_device_scalars)
```

Comment:
Added a memory pool to the tests, since the grudge `PyOpenCLArrayContext` recommended it. Not sure if this is a good idea?

Reply:
That should be OK, as long as that array context gets freed.